lib/std/debug.zig

```diff
@@ -563,7 +563,7 @@ fn machoSearchSymbols(symbols: []const MachoSymbol, address: usize) ?*const Mach
     while (min < max) {
         const mid = min + (max - min) / 2;
         const curr = &symbols[mid];
-        const next = &symbols[mid + 1];
+        const next = &symbols[@minimum(mid + 1, symbols.len - 1)];
```

Comment:
Out of curiosity, did you check if the same effect can be achieved by initially setting `max = len - 1` instead of `max = len`?

Reply:
IIRC I did try that and it didn't work for some reason, let me try again to confirm 👍

Reply:
Seems like this does work, probably I tried it before I added the second half of the fix. Updated 👍